deploy/eck-operator/templates/service-monitor.yaml

```diff
@@ -19,10 +19,13 @@ spec:
     scheme: https
     interval: 30s
     tlsConfig:
-      insecureSkipVerify: {{ .Values.config.metrics.secureMode.tls.insecureSkipVerify | default false }}
-      {{- if (not .Values.config.metrics.secureMode.tls.insecureSkipVerify) }}
-      {{- $leading_path := trimSuffix "/" .Values.config.metrics.secureMode.tls.caMountDirectory }}
-      {{- with .Values.config.metrics.secureMode.tls.caSecret }}
+      {{- $insecureSkipVerify := (ternary .Values.config.metrics.secureMode.tls.insecureSkipVerify .Values.serviceMonitor.insecureSkipVerify (hasKey .Values.config.metrics.secureMode.tls "insecureSkipVerify")) }}
+      insecureSkipVerify: {{ $insecureSkipVerify }}
+      {{- if (not $insecureSkipVerify) }}
+      {{- $caMountDirectory := or (.Values.config.metrics.secureMode.tls.caMountDirectory) (.Values.serviceMonitor.caMountDirectory) -}}
```

Comment:
I don't see `.Values.config.metrics.secureMode.tls.caMountDirectory` as a valid option any longer after your changes @pebrc

Reply:
I left this in place in case users have old values files with this structure that I did not want to break.